username.rb

```diff
@@ -2,13 +2,73 @@
 # Run `rspec /spec/username_spec.rb` to get started.
 
 def format_name(first, last)
-  nil
+
+      if (first == "" || last == "")
+      nil
+      else
+      first_name = (first.gsub /\s/, '')[0] + last
+      first_name.downcase.gsub /\s/, ''
+      end
 end
 
-def format_year
+def format_year(year)
+
+  if (year.to_s.length !=  4)
   nil
+  else
+  year.to_s[2,3]
+  end
+  end
+
+def check_privilege(i = 0)
+a = i.floor
+  if (a == 0)
+    "user"
+  elsif (a == 1)
+    "seller"
+  elsif (a == 2)
+    "manager"
+  elsif (a >= 3)
+    "admin"
+  else
+    nil
+  end
+
 end
 
-def build_username
-  nil
+def user_type_prefix(i)
+  if (i == 0)
+    ""
+  else
+  check_privilege(i) + "-"
+end
+
+end
+
+def build_username(first, last, year, i = 0)
+
+  user_type_prefix(i) + format_name(first, last) + format_year(year)
+end
+
+def generate_username(first, last, year, i = 0)
+  # a = [build_username]
+  # indexof  the new username in the array of stored usernames if it's greater than or == to 0
+
+  a = build_username(first, last, year, i)
```

Comment:
Did this work for you? I think you're close but I don't know this would work as written.